geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreFactory.scala

```diff
@@ -104,45 +103,60 @@ object AccumuloDataStoreFactory extends GeoMesaDataStoreInfo {
       null,
       Map(Parameter.DEPRECATED -> true, Parameter.IS_PASSWORD -> true).asJava)
 
-  override def canProcess(params: java.util.Map[String, _ <: Serializable]): Boolean = {
-    val hasConnection = InstanceIdParam.exists(params) && ZookeepersParam.exists(params) && UserParam.exists(params)
-    // exactly one of password or keytab
-    hasConnection && (PasswordParam.exists(params) != KeytabPathParam.exists(params))
-  }
+  override def canProcess(params: java.util.Map[String, _ <: Serializable]): Boolean =
+    CatalogParam.exists(params)
```

Comment:
Would anything depend on the particular behavior of the canProcess method?  (I'm wondering how big a deal this change would be.)

I think the answer should be no, but just wanted to discuss it.

Reply:
i don't think so... i think that having `canProcess` be more lenient and then having `createDataStore` more strict results in better error messages. E.g. before:
```
scala> DataStoreFinder.getDataStore(Map("foo" -> "bar"))
res1: org.geotools.data.DataStore = null
scala> DataStoreFinder.getDataStore(Map("accumulo.catalog" -> "bar"))
res2: org.geotools.data.DataStore = null
```
and after:
```
scala> DataStoreFinder.getDataStore(Map("accumulo.catalog" -> "bar"))
Apr 13, 2020 5:54:22 PM org.geotools.data.DataAccessFinder getDataStore
WARNING: Accumulo (GeoMesa) should be used, but could not connect
java.io.IOException: Parameter accumulo.instance.id is required: Accumulo Instance ID
	at org.locationtech.geomesa.accumulo.data.AccumuloDataStoreFactory$$anonfun$lookup$1$1.apply(AccumuloDataStoreFactory.scala:116)
<more stack trace>
java.io.IOException: Parameter accumulo.instance.id is required: Accumulo Instance ID
  at org.locationtech.geomesa.accumulo.data.AccumuloDataStoreFactory$$anonfun$lookup$1$1.apply(AccumuloDataStoreFactory.scala:116)
<more stack trace>
```

Reply:
i think technically this breaks the data store API, but seems better really. What do you think?

Reply:
I don't know what the exact API is.  I think this is sensible.  

A 'true' result from canProcess doesn't promise that the connection will work.  Throwing IllegalArgumentExceptions seems rather reasonable here.  (I don't know when IOExceptions or IllegalArgumentExceptions make more sense in a situation like this.)